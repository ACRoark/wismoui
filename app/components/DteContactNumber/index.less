@import (reference) '../../styles/themes/index.less';

@dte-contact-number-cls-prefix: ~'@{dte-prefix}-dte-contact-number';

.@{dte-contact-number-cls-prefix} {
  color: @dte-blue-light;
  font-weight: @dte-wismo-font-weight-semi-bold;
  padding-left: 2px !important;

  @media @mobile {
    text-decoration: underline;
  }
}
