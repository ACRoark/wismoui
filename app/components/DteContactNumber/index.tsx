import * as React from 'react';

import './index.less';

interface DteContactNumberProps {
  value?: string;
}

const DteContactNumber: React.FC<DteContactNumberProps> = (props: DteContactNumberProps): React.ReactElement => {
  const { value } = props;
  const contactNumber = value || '8004774747';

  return (
    <a className="dte-wismo-dte-contact-number" href={`tel:+${contactNumber}`}>
      {`${contactNumber.slice(0, 3)}.${contactNumber.slice(3, 6)}.${contactNumber.slice(6, 10)}`}
    </a>
  );
};

export default DteContactNumber;
