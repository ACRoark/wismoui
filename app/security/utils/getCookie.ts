const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
