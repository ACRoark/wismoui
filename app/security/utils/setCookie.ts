interface ICookieOptions {
  days: number;
  path: string;
}

const setCookie = (name: string, value: string, options: ICookieOptions): void => {
  const expires = new Date(Date.now() + options.days * 864e5).toUTCString();

  document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=${options.path}`;
};

export default setCookie;
