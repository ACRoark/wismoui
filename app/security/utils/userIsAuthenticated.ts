import useAuth from 'hooks/useAuth';

const userIsAuthenticated = (): boolean => {
  const { isAuthenticated } = useAuth();

  return isAuthenticated;
};

export default userIsAuthenticated;
