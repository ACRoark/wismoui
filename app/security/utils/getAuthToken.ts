import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
