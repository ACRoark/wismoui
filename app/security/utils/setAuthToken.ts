import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
