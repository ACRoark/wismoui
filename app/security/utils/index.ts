import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
