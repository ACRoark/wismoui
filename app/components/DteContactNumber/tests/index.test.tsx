import { render } from 'enzyme';
import React from 'react';

import { createSnapshot } from 'testing/utils';

import DteContactNumber from '..';

const expectComponentToMatchSnapshot = (value?: string): void => {
  const tree = createSnapshot(<DteContactNumber value={value} />);

  expect(tree).toMatchSnapshot();
};

describe('<DteContactNumber />', (): void => {
  it('should not log errors in console when rendered', (): void => {
    const spy = jest.spyOn(global.console, 'error');

    render(<DteContactNumber />);

    expect(spy).not.toHaveBeenCalled();
  });

  it('should render correctly when value is 8004774747', (): void =>
    expectComponentToMatchSnapshot('8004774747'));
});
