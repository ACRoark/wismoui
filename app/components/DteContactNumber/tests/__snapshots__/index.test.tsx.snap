// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DteContactNumber /> should render correctly when value is 8004774747 1`] = `
<a
  className="dte-wismo-dte-contact-number"
  href="tel:+8004774747"
>
  800.477.4747
</a>
`;
